packages/web/src/shared/hooks/useAuth.ts

```diff
@@ -0,0 +1,67 @@
+import { useQuery } from '@tanstack/react-query';
+import { GetSessionResponse, SystemRole } from '@translation/api-types';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import apiClient from '../apiClient';
+
+export interface UseAuthOptions {
+  authenticated?: boolean;
+  systemRoles?: SystemRole[];
+}
+
+export type UseAuthResult =
+  | {
+      status: 'authenticated';
+      user: Required<GetSessionResponse>['user'];
+    }
+  | { status: 'unauthenticated' | 'loading'; user?: undefined };
+
+export default function useAuth({
+  authenticated,
+  systemRoles,
+}: UseAuthOptions = {}): UseAuthResult {
+  const navigate = useNavigate();
+  const { data, status } = useQuery(['session'], async () =>
+    apiClient.getSession()
+  );
+
+  useEffect(() => {
+    if (status === 'success') {
+      const { user } = data;
+      if (user) {
+        if (
+          authenticated === false ||
+          (systemRoles &&
+            !systemRoles.some((role) => user.systemRoles.includes(role)))
+        ) {
```

Comment:
I'm not getting the `authenticated` usage here and line 43.

Reply:
Ahhh, I might be understanding now. Need better explanation comments.

Reply:
I added explanation comments to the best of my understanding. Please improve them if they are incorrect. I need additional clarification about the `authenticated === false` on line 35.

Reply:
I rewrote the hook so that it's easier to understand both the options and the flow of logic. I didn't like that the name of the `authenticated` variable didn't reveal it's purpose without a comment. I also added a jsdoc comment to the hook itself that explains how to use it. Let me know if something is still missing.